ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/admin/identity_management_service.proto

```diff
@@ -0,0 +1,138 @@
+
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.digitalasset.ledger.api.v1.admin;
+
+option java_outer_classname = "PartyManagementServiceOuterClass";
+option java_package = "com.digitalasset.ledger.api.v1.admin";
+
+// Status: experimental interface, will change before it is deemed production
+// ready
+
+// Inspect the identity management state of a ledger participant and modify the
+// parts that are modifiable. We use 'backing participant' to refer to this
+// specific participant in the methods of this API.
+// It is not mandatory to implement this interface on a specific daml-on-x
+// implementation or to implement it in its entirety
+service IdentityManagementService {
+
+  // Return the identifier of the backing participant.
+  // All horizontally scaled replicas should return the same id.
+  // daml-on-sql: returns an identifier supplied on command line at launch time
+  // daml-on-sawtooth: as above
+  // canton: returns globally unique identifier
+  rpc GetParticipantId (GetParticipantIdRequest) returns
+    (GetParticipantIdResponse);
+
+  // List the participants known by the backing participant.
+  // TODO: Check what the use case is for listing all participants
+  // Admin tooling in daml-on-sawtooth knows the topology from other sources
+  // daml-on-sql knows only one participant. Suggest to remove it
+  rpc ListKnownParticipants (ListKnownParticipantsRequest) returns
+    (ListKnownParticipantsResponse);
+
+  // List the parties known by the backing participant.
+  // The list returned contains parties whose ledger access is facilitated by
+  // backing participant and teh ones maintained elsewhere
+  // TODO: Check what the use case is for listing non-local parties
+  rpc ListKnownParties (ListKnownPartiesRequest) returns
+    (ListKnownPartiesResponse);
+
+  // Adds a new party to the set managed by the backing participant.
+  // Caller specifies a party identifier suggestion, the actual identifier
+  // allocated might be different and is implementation specific.
+  // daml-on-sql: suggestion's uniqueness is checked and call rejected if the
+  // identifier is already present
+  // daml-on-sawtooth: suggestion is prefixed with backing participant
```

Comment:
Let's not use the backing participant identifier. The KVutils will maintain the proper mapping to implement the uniqueness check as in the daml-on-sql case. This way more freedom is provided, and better demos can be done.

Reply:
OK, I will update the partners about this change. I should work with the validators making sure no duplicates or appropriation of used parties by another participant is happening.